### PR TITLE
feat(feature tour): FF for hero

### PR DESCRIPTION
**Pull request by kishore03109**
## Problem

We are releasing new homepage features. [figma link](https://www.figma.com/file/7OtNYkc4s9BkFk05ConEbd/%E2%AD%90-V1.5-Master?type=design&node-id=14663-161877&mode=dev)

Closes [insert issue #]

## Solution

This PR implements the feature tour + storybooks for this. Feature tour is only shown if feature flag for using the new `HOMEPAGE_TEMPLATES` are enabled 

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes

- [X] No - this PR is backwards compatible
- While API contract for feature tour changed, i preserved the default behaviour of the components. 

## Screenshots

Screenshot 2023-09-20 at 4 39 32 PM

https://github.com/isomerpages/isomercms-frontend/assets/42832651/f7da254b-0398-4f1a-a4c0-f2ff2eec9a58

## Blocker (potential chrome bug in local machine only)
For some weird reason, the spotlight offsets to create this weird effect. Notice that the tooltip renders properly, but not the spotlight. I did find a related issue [here](https://github.com/gilbarbara/react-joyride/issues/376), but the issue was fixed so it seems unlikely to be the cause. NOTE: This was only reproducible in chrome, and only in my local machine. :sadge: 

Screenshot 2023-09-25 at 8 38 05 AM

Interestingly, after hot refreshes, the spotlight magically renders to the correct location. I thought this is suggestive of the need to render the spotlight AFTER the render of the Text, but somehow using a useEffect did not solve this :(
Screenshot 2023-09-25 at 9 05 27 AM

## Tests

- [ ] Ensure that you are a whitelisted site 
- [ ] Login enter in to the `/homepage` site. 
- [ ] Ensure that when you expand the hero section, feature tour pops up

**New dependencies**:

- `dependency` : react-intersection-observer
We need to to check if an element is in view. Else feature tour doesnt work, with this error thrown in the console. 
Screenshot 2023-09-20 at 4 06 42 PM

### CODE REVIEW

File: src/layouts/components/Homepage/HeroDropdownSection.tsx [hunk 1]
```diff
@@ -31,7 +31,10 @@ interface HeroDropdownSectionProps extends EditorHeroDropdownSection {
 
 export const HeroDropdownSection = ({
   errors,
-  dropdown,
+  dropdown = {
+    options: [],
+    title: "", //! todo: check how Chin handled this error after rebase
```
Comment (kishore03109) [inline src/layouts/components/Homepage/HeroDropdownSection.tsx:36]: Blocker here! can ignore this for now!

**Comment by kishore03109** [comment]
!run chromatic

File: src/features/FeatureTour/FeatureTourTooltip.tsx [hunk 2]
```diff
@@ -30,32 +30,37 @@ export const FeatureTourTooltip = ({
...
 }: FeatureTourTooltipProps): JSX.Element => {
   const { paginationCallback } = useFeatureTourContext()
+  const showProgressIndicator = size > 1
   return (
     <Box
       padding="1.5rem"
       alignItems="center"
       maxW="100%"
-      w="18rem"
+      w="20rem"
```
Comment (kishore03109) [inline src/features/FeatureTour/FeatureTourTooltip.tsx:42]: Confirmed wit Design to standardise this already

File: src/features/FeatureTour/FeatureTourTooltip.tsx [hunk 3]
```diff
@@ -68,14 +73,16 @@ export const FeatureTourTooltip = ({
...
+          <ProgressIndicator
+            numIndicators={size}
+            currActiveIdx={index}
+            onClick={paginationCallback}
+          />
+        )}
         {isLastStep ? (
           <Button {...primaryProps} title="Done">
-            Done
+            Got it
```
Comment (kishore03109) [inline src/features/FeatureTour/FeatureTourTooltip.tsx:85]: Confirmed wit Design to standardise this already

**Comment by kishore03109** [comment]
!run chromatic

**Comment by kishore03109** [comment]
!run chromatic

File: src/layouts/components/Homepage/HeroBody.tsx [hunk 4]
```diff
@@ -307,45 +307,52 @@ const HeroLayoutForm = ({
...
-          steps={HERO_OPTIONS_FEATURE_STEPS}
-        />
+        /**
+         * Because of the VStack, this renders as a valid child with padding
+         * which messes up the layout. The caller of this method also wraps
+         * this around a VStack, so placing the FeatureTourHandler outside of this
+         * parent VStack does not solve the issue. As a workaround, the margin
+         * top is set to -1 to 'undo' the spacing by VStack.
+         */
+        <Box mt="-1rem">
```
Comment (kishore03109) [inline src/layouts/components/Homepage/HeroBody.tsx:319]: Yes this is ugly, but I am not sure how to quickly fix this without modifying a huge chunk of the code. 
### Before
Screenshot 2023-09-22 at 3 19 55 PM

### After
Screenshot 2023-09-22 at 3 19 36 PM

**Comment by harishv7** [comment]
@kishore03109 unable to replicate blocker. Might want to de-prio if this is not happening on staging build

**Review by harishv7 — APPROVED** [review]

**Pull request merged**